### PR TITLE
restore monolith example

### PR DESCRIPTION
**Pull request by joshgav**
This restores the original "podtatoserver" single-service project as `/podtato-head-monolith` and adds build and test scripts for it at `build/build_image.sh` and `build/test_image.sh`. Some examples - like keptn, see #139 - are based on this older project so we are considering restoring it at least for now.

This PR mostly restores the monolith project as it was prior to #129 and moves the microservices project to `/podtato-head-microservices`. If we'd like to keep the "monolith" example around permanently, there's opportunity to document it a bit and add the build/test scripts to the repo's Makefile and GitHub workflows.

### CODE REVIEW

File: podtato-head-monolith/podtatoserver.go
Comment (TannerGabriel) [inline podtato-head-monolith/podtatoserver.go:140]: After testing this out it did not expose the `static` directory with the absolute path (`/static'`) in the `http.Dir` command. Changing it to a relative path (`./static`) worked:

```
Handler(http.StripPrefix(staticDir, http.FileServer(http.Dir("."+staticDir))))
```

File: podtato-head-monolith/podtatoserver.go
Comment (TannerGabriel) [inline podtato-head-monolith/podtatoserver.go:132]: The directory can be changed here, but `/static` is hard-coded in other occurrences like line 76 and therefore changing the directory from `/static` would break the code.

File: podtato-head-monolith/podtatoserver.go
Comment (TannerGabriel) [inline podtato-head-monolith/podtatoserver.go:84]: Maybe change this delay to something smaller e.g. 400 milliseconds to reduce waiting time for tests with 1000+ requests.

File: podtato-head-monolith/docker/DockerfileV0.1.2
Comment (TannerGabriel) [inline podtato-head-monolith/docker/DockerfileV0.1.2:1]: We could combine these 3 Dockerfiles into one and use an `arg` to set the building version.

```
docker build --build-arg VERSION=0.1.2 .
```

Only an idea to reduce code duplication.

**Review by TannerGabriel — CHANGES_REQUESTED** [review]

File: podtato-head-monolith/static/overview.html
Comment (TannerGabriel) [inline podtato-head-monolith/static/overview.html:1]: Is this file even used?

File: podtato-head-monolith/docker/DockerfileV0.1.0
Comment (TannerGabriel) [inline podtato-head-monolith/docker/DockerfileV0.1.0:28]: The HTML file is copied to `/podtato-new.html` but the script still uses `/static/podtato-new.html`. The line could therefore be deleted.

**Review by TannerGabriel — CHANGES_REQUESTED** [review]

File: podtato-head-monolith/podtatoserver.go
Comment (joshgav) [reply podtato-head-monolith/podtatoserver.go:140]: Interesting! I had problems with the existing default prefix `./static/` and only got the test (`build/test_image.sh`) to work with `/static`. _That_ works cause in the Dockerfile the binary itself and the `static` dir are copied to the root (`/`); it wouldn't work if running the binary directly though.

One problem in the code is that it uses the same string variable to specify the path prefix in the URL _and_ the path to find the files on the file system; let me see about changing that.

This also motivates me to add a proper GitHub workflow to build the image and run the test so we can ensure it works the same for everyone.

File: podtato-head-monolith/podtatoserver.go
Comment (joshgav) [reply podtato-head-monolith/podtatoserver.go:132]: Thanks! I'll take a look at that. Note that I didn't write this code, I'm really just proposing to restore it as is with a couple tweaks to make tests pass.

File: podtato-head-monolith/podtatoserver.go
Comment (joshgav) [reply podtato-head-monolith/podtatoserver.go:84]: A reasonable idea, but I think it should be in a follow-on PR so we can keep things here as close as possible to how they were originally.

File: podtato-head-monolith/docker/DockerfileV0.1.2
Comment (joshgav) [reply podtato-head-monolith/docker/DockerfileV0.1.2:1]: Makes sense to me. Again, perhaps we can merge in as things were previously and defer enhancements to follow-on PRs.

File: podtato-head-monolith/static/overview.html
Comment (joshgav) [reply podtato-head-monolith/static/overview.html:1]: :man_shrugging: :smile: Same as others, let's analyze if and after we merge.

File: podtato-head-monolith/podtatoserver.go
Comment (joshgav) [reply podtato-head-monolith/podtatoserver.go:140]: I added a GitHub workflow to build and test the images using the scripts I added and it passes: <https://github.com/podtato-head/podtato-head/actions/runs/1654869776>.

Not sure why this code doesn't work for you @TannerGabriel! Are you running the binary outside of the image perhaps?

Comment (TannerGabriel) [reply podtato-head-monolith/podtatoserver.go:140]: A proper Github workflow sounds excellent!

File: podtato-head-monolith/podtatoserver.go
Comment (TannerGabriel) [reply podtato-head-monolith/podtatoserver.go:132]: Yes, I know it is not your codebase. Thanks again for reinstating this code. These are only some things that I noticed when looking over the code.

File: podtato-head-monolith/podtatoserver.go
Comment (TannerGabriel) [reply podtato-head-monolith/podtatoserver.go:84]: Yes that sounds reasonable.

File: podtato-head-monolith/podtatoserver.go
Comment (joshgav) [reply podtato-head-monolith/podtatoserver.go:140]: Added in latest commit, PTAL!

Comment (joshgav) [reply podtato-head-monolith/podtatoserver.go]: In the ensuing commit I distinguished the file system path from the URL path prefix for static assets. Now the filesystem path is `./static` and the URL path prefix is `/static`, which seems correct. Tests pass. Can you try it @TannerGabriel?

File: podtato-head-monolith/podtatoserver.go
Comment (joshgav) [reply podtato-head-monolith/podtatoserver.go]: Fixed the issue in line 76 in the latest commit, look good to you? I didn't see other code with static `static/` (sic :smile:) to fix at the moment.

File: podtato-head-monolith/docker/DockerfileV0.1.0
Comment (joshgav) [reply podtato-head-monolith/docker/DockerfileV0.1.0]: Removed this in the latest commit.

**Review by TannerGabriel — APPROVED** [review]
LGTM

File: podtato-head-monolith/podtatoserver.go
Comment (TannerGabriel) [reply podtato-head-monolith/podtatoserver.go]: Yes, that worked for me!

**Comment by TannerGabriel** [comment]
@joshgav This PR is mergeable from my point of view. Are there any changes required from your point?

**Comment by joshgav** [comment]
Quick update: the previous commits weren't pushing built images to GHCR yet so the latest commits are to aid with that, but they don't work yet - negotiating GHCR permissions is a really big pain. I'm working on it but may not get to it today.

**Comment by TannerGabriel** [comment]
@joshgav thank you for the effort. Time is not really a big problem. I only asked because I wanted to know if this PR needed any action on my site. Let me know if you need any help or another review.

**Pull request merged**
